core/app/models/spree/product.rb

```diff
@@ -323,7 +323,7 @@ def taxons_for_store(store)
     end
 
     def any_variant_in_stock_or_backorderable?
-      only_master_variant = variants_including_master.count == 1 && variants.count.zero?
+      only_master_variant = variants_including_master.count == 1 && variants.none?
```

Comment:
```suggestion
      only_master_variant = variants.none?
```

we don't need to count variants, master variant is always created